### PR TITLE
No more saml engine hub certs

### PR DESCRIPTION
**Pull request by rhowe-gds**
SAML Engine doesn't need the hub SAML signing & encryption certificates, just the private keys.

Remove them from the configuration and remove the code that handles them.

The encryption certificates were already ignored since at least December 2017.

Also, switch to using the OpenSAML helper function for deriving a public/private keypair from a private key and remove our homegrown implementation.

### CODE REVIEW

File: hub/saml-engine/src/main/java/uk/gov/ida/hub/samlengine/SamlEngineModule.java [hunk 1]
```diff
@@ -443,15 +444,15 @@ private JsonClient jsonClient(JsonResponseProcessor jsonResponseProcessor, Envir
...
+            KeyPair signingKeyPair = new KeyPair(KeySupport.derivePublicKey(privateSigningKey(configuration)), privateSigningKey(configuration));
 
-        KeyPair primaryEncryptionKeyPair = Crypto.keyPairFromPrivateKey(privateKeyStore.get(KeyPosition.PRIMARY));
-        KeyPair secondaryEncryptionKeyPair = Crypto.keyPairFromPrivateKey(privateKeyStore.get(KeyPosition.SECONDARY));
-        KeyPair signingKeyPair = Crypto.keyPairFromPrivateKey(privateSigningKey(configuration));
-
-        return new IdaKeyStore(signingCertificate, signingKeyPair, asList(primaryEncryptionKeyPair, secondaryEncryptionKeyPair));
+            return new IdaKeyStore(signingKeyPair, asList(primaryEncryptionKeyPair, secondaryEncryptionKeyPair));
+        } catch (KeyException e) {
+            throw new RuntimeException(e);
```
Comment (rhowe-gds) [inline hub/saml-engine/src/main/java/uk/gov/ida/hub/samlengine/SamlEngineModule.java]: So sue me

File: hub/saml-engine/src/main/java/uk/gov/ida/hub/samlengine/SamlEngineModule.java [hunk 2]
```diff
@@ -440,18 +439,18 @@ private JsonClient jsonClient(JsonResponseProcessor jsonResponseProcessor, Envir
 
     @Provides
     @Singleton
-    private IdaKeyStore getKeyStore(X509CertificateFactory certificateFactory, SamlEngineConfiguration configuration) {
+    private IdaKeyStore getKeyStore(SamlEngineConfiguration configuration) {
```
Comment (rhowe-gds) [inline hub/saml-engine/src/main/java/uk/gov/ida/hub/samlengine/SamlEngineModule.java]: So.. tell me about guice and what the impact of this is

File: hub/saml-engine/src/main/java/uk/gov/ida/hub/samlengine/SamlEngineModule.java [hunk 3]
```diff
@@ -440,18 +439,21 @@ private JsonClient jsonClient(JsonResponseProcessor jsonResponseProcessor, Envir
...
```
[the remark this answers is not in the document]
Comment (rhowe-gds) [reply hub/saml-engine/src/main/java/uk/gov/ida/hub/samlengine/SamlEngineModule.java]: I don't think you're qualified to comment, codacy

File: hub/saml-engine/src/main/java/uk/gov/ida/hub/samlengine/SamlEngineModule.java (hunk 2 above)
Comment (willp-bl) [reply hub/saml-engine/src/main/java/uk/gov/ida/hub/samlengine/SamlEngineModule.java]: there isn't one :D

**Pull request merged**